x-pack/plugins/security_solution/server/lib/detection_engine/rules/saved_object_mappings.ts

```diff
@@ -9,38 +9,84 @@ import { SavedObjectsType } from '../../../../../../../src/core/server';
 
 export const ruleStatusSavedObjectType = 'siem-detection-engine-rule-status';
 
-export const ruleStatusSavedObjectMappings: SavedObjectsType['mappings'] = {
+/**
+ * Records monitoring information about a rule, such as a when it last failed and why.
+ * This object is separate of the Rule SavedObject (why?)
```

Comment:
@dhurley14 can correct me here, but I believe the two main motivations here were:

* Updating the alert SO itself was problematic as it could invalidate API keys (I don't believe this is still the case)
* The desire to persist multiple statuses
  * The idea of simply persisting an array on the Alert may have been discussed, but I believe we wanted to avoid an ever-growing Alert SO.

Reply:
@rylnd is correct. Also, I'm pretty sure it is still the case that we cannot update the alert while it is executing. At least at the time of developing this, if that was the possible we would have kept all of the status data in the params of the rule's SO.